### PR TITLE
Add support for ROS Noetic

### PR DESCRIPTION
**Pull request by gonzalocasas**
There were some minor tweaks required for the moveit planner to be compatible with ROS Noetic.

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.rst` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas_fab.robots.Configuration`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_fab/backends/ros/client.py [hunk 1]
```diff
@@ -151,6 +153,14 @@ def __enter__(self):
     def __exit__(self, *args):
         self.close()
 
+    @property
+    def ros_distro(self):
+        """Retrieves the ROS version to which the client is connected (eg. kinetic)"""
+        if not self._ros_distro:
+            self._ros_distro = Param(self, '/rosdistro').get(timeout=3)
```
Comment (beverlylytle) [inline src/compas_fab/backends/ros/client.py:160]: woah, the distro retrieval gets a 3 second timeout, but the default for ik is only 1?

Comment (gonzalocasas) [reply src/compas_fab/backends/ros/client.py]: hehe.. good point! made them more coherent on https://github.com/compas-dev/compas_fab/pull/279/commits/f8192e9757e968b7b8714165232969b3d38150a1

File: src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py [hunk 2]
```diff
@@ -108,12 +111,22 @@ def inverse_kinematics_async(self, callback, errback,
...
+        timeout_duration = Duration(timeout_in_secs, 0).to_data()
+
         ik_request = PositionIKRequest(group_name=group,
                                        robot_state=start_state,
                                        constraints=constraints,
                                        pose_stamped=pose_stamped,
                                        avoid_collisions=options.get('avoid_collisions', True),
-                                       attempts=options.get('attempts', 8))
+                                       attempts=options.get('attempts', 8),
+                                       timeout=timeout_duration)
```
Comment (beverlylytle) [inline src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py:123]: Two questions here.  Why the `Duration.to_data()`?  Secondly, the default value for `PositionIKRequest` is a float, which doesn't seem consistent with the ros documentation that says it should be a `Duration`.

Comment (beverlylytle) [reply src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py:123]: aha, i see that `Duration` is not a `ROSmsg`, hence the `to_data`

Comment (gonzalocasas) [reply src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py:123]: good catch, the default `timeout` in `PositionIKRequest` was invalid (I wonder why it worked at all, maybe there's an automatic conversion between float and `Duration`.

Comment (gonzalocasas) [reply src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py:123]: fixed default on https://github.com/compas-dev/compas_fab/pull/279/commits/8f2fedcb7f227fbcef7fa561c3cbd25a363dd1b9

File: src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py [hunk 3]
```diff
@@ -108,12 +111,22 @@ def inverse_kinematics_async(self, callback, errback,
...
+
+        # The field `attempts` was removed in Noetic (and higher)
+        # so it needs to be removed from the message otherwise it causes a serialization error
+        # https://github.com/ros-planning/moveit/pull/1288
+        if self.ros_client.ros_distro not in ('kinetic', 'melodic'):
```
Comment (beverlylytle) [inline src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py:128]: It feels like there should be a `Ros_distro` enum.

**Review by beverlylytle — APPROVED** [review]
LGTM!

File: src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py (hunk 3 above)
Comment (gonzalocasas) [reply src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py]: added a minimal one on https://github.com/compas-dev/compas_fab/pull/279/commits/032799c424c5eee23d148235a695571e25835d95

**Comment by jf---** [comment]
@gonzalocasas hmmm, is this a very tasteful way of celebration of [`conda install ros-noetic-moveit`](https://medium.com/robostack/cross-platform-conda-packages-for-ros-fa1974fd1de3)?

**Comment by gonzalocasas** [comment]
@jf--- timing was only a happy coincidence, but just tested it and it works like a charm! 🎉
image

**Comment by jf---** [comment]
Unbelievable!!

**Pull request merged**